src/rebar_app_discover.erl

```diff
@@ -56,6 +57,27 @@ merge_deps(AppInfo, State) ->
 
     {AppInfo1, State1}.
 
+project_app_config(AppInfo, State) ->
+    C = rebar_config:consult(rebar_app_info:dir(AppInfo)),
+    Dir = rebar_app_info:dir(AppInfo),
+    maybe_reset_hooks(C, Dir, State).
+
+%% Here we check if the app is at the root of the project.
+%% If it is, then drop the hooks from the config so they aren't run twice
+maybe_reset_hooks(C, Dir, State) ->
+    case filename:dirname(rebar_dir:root_dir(State)) of
+        Dir ->
+            C1 = proplists:delete(provider_hooks, C),
+            proplists:delete(hooks, C1);
+        _ ->
+            C
+    end.
+
+reset_hooks(State) ->
+    lists:foldl(fun(Key, StateAcc) ->
+                        rebar_state:set(StateAcc, Key, [])
+                end, State, [post_hooks, pre_hooks, provider_hooks]).
+
```

Comment:
Doesn't this function need to care about the directory the same way `maybe_reset_hooks` did?